### PR TITLE
Rotation / Architecture for Drop-in UI 

### PR DESCRIPTION
**Pull request by danesfeder**
Introduces architecture to the drop-in UI for rotation.
Found in #223 `NavigationView` was responsible for everything (got bulky fast).
This removes all location / nav / route logic from `NavigationView` into their own `ViewModel` and creates a presenter to handle logic / updates from the `ViewModels`

Todo: 
- [x] startup with coordinates
- [x] actually rotate
- [x] landscape views
- [x] camera resume on location
- [ ] javadoc

### CODE REVIEW

**Comment by danesfeder** [comment]
The puck resets in this because this is with the `MockLocationEngine` (which always starts from the beginning of the route) 

ezgif com-video-to-gif

File: navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationContract.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package com.mapbox.services.android.navigation.ui.v5;
+
+import com.mapbox.services.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.services.commons.models.Position;
+
+public interface NavigationContract {
+
+  interface View {
```
Comment (tobrun) [inline navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationContract.java]: at some point, no requirement for this PR, we could look into decoupling this interface into smaller interfaces eg. SummaryView which encapsulates the methods below or a RecenterView that encapsulates hide/show.

File: navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+public class NavigationViewModel extends AndroidViewModel implements LifecycleObserver, ProgressChangeListener,
+  MilestoneEventListener, OffRouteListener {
+
+  public MutableLiveData<InstructionModel> instructionModel = new MutableLiveData<>();
+  public MutableLiveData<SummaryModel> summaryModel = new MutableLiveData<>();
+  public MutableLiveData<Boolean> isOffRoute = new MutableLiveData<>();
+  MutableLiveData<Location> navigationLocation = new MutableLiveData<>();
+  MutableLiveData<Position> newOrigin = new MutableLiveData<>();
+  MutableLiveData<Boolean> isRunning = new MutableLiveData<>();
```
Comment (tobrun) [inline navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java]: if possible make above private and final.

File: navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationContract.java (hunk 1 above)
Comment (danesfeder) [reply navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationContract.java]: @tobrun Yeah definitely, would you recommend each custom view having their own presenter as well?

**Comment by danesfeder** [comment]
Javadoc will be added in a separate PR

**Pull request merged**